core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -117,11 +118,15 @@ class SearchService(
     // index all the given bases and return number of rows written
     def indexBases(bases: Set[FileObject], checks: Seq[FileCheck]): Future[Int] = {
       log.info("Indexing bases...")
-      val checksLookup: Map[String, FileCheck] = checks.map(check => (check.filename -> check)).toMap
-      val basesWithChecks: Set[(FileObject, Option[FileCheck])] = bases.map { base =>
-        (base, checksLookup.get(base.getName().getURI()))
+      val known: Map[FileObject, FileCheck] = checks.flatMap { check =>
+        val f = check.file
+        if (bases(f)) Some(check.file -> check)
```

Comment:
It's not clear to me that FileObject does implement equals properly. Looking at the source for [AbstractFileObject](http://grepcode.com/file/repo1.maven.org/maven2/org.apache.commons/commons-vfs2/2.0/org/apache/commons/vfs2/provider/AbstractFileObject.java#AbstractFileObject) and [ZipFileObject](http://grepcode.com/file/repo1.maven.org/maven2/org.apache.commons/commons-vfs2/2.0/org/apache/commons/vfs2/provider/zip/ZipFileObject.java?av=f) it doesn't seem like it's implemented. If it inherits from `Object`, is the map always going to be empty?